tests/unit/sstring_test.cc

```diff
@@ -199,6 +199,27 @@ BOOST_AUTO_TEST_CASE(test_append) {
     BOOST_REQUIRE_EQUAL(sstring("aba").append("1234", 0), "aba");
 }
 
+BOOST_AUTO_TEST_CASE(test_push_back) {
+    {
+        // append in internal storage
+        basic_sstring<char, uint32_t, 15> s("0123456789");
+        s.push_back('a');
+        BOOST_REQUIRE_EQUAL(s, "0123456789a");
+    }
+    {
+        // append causing spilling to external storage
```

Comment:
I would do this test in a loop, for many sizes, not just 15 - e.g. from size 0 to 50. I don't know if 15 exactly is the cutoff point, or maybe 14 or 16? I'm worried that the above code has multiple bugs (I pointed out my suspicions above) and this test doesn't make me confident.
Also please check the NulTerminate option, whatever it is.